invenio_records_resources/services/records/results.py

```diff
@@ -511,3 +506,77 @@ def expand(self, identity, hit):
                 dict_merge(results, d)
 
         return results
+
+
+class MultiFieldsResolver(FieldsResolver):
+    """Resolve the reference record for each of the configured fields.
+
+    Given a list of fields referencing other records/objects,
+    it fetches and returns the dereferenced record/obj.
+
+    This class supports resolution of nested fields and efficiently batches
+    resolution requests to services.
+    """
+
+    def _collect_values(self, hits):
```

Comment:
nit: If there are any common code blocks here I would extract them in the base class so we can reuse them here and keep only the needed modified parts to support the list of things